nym-vpn-core/crates/nym-vpn-account-controller/src/storage/credentials/pending_credential_requests/mod.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+pub mod error;
+pub mod models;
+
+mod sqlite;
+
+use std::{
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use sqlite::SqliteZkNymRequestsStorageManager;
+use sqlx::ConnectOptions;
+use tracing::log::LevelFilter;
+
+use error::PendingCredentialRequestsStorageError;
+use models::{PendingCredentialRequest, PendingCredentialRequestStored};
+
+#[derive(Clone)]
+pub(crate) struct PendingCredentialRequestsStorage {
+    storage_manager: SqliteZkNymRequestsStorageManager,
+    database_path: PathBuf,
+}
+
+impl PendingCredentialRequestsStorage {
+    pub(crate) async fn init<P: AsRef<Path>>(
+        database_path: P,
+    ) -> Result<Self, PendingCredentialRequestsStorageError> {
+        tracing::info!(
+            "Setting up pending credential requests storage: {:?}",
+            database_path.as_ref().as_os_str()
+        );
+
+        let opts = sqlx::sqlite::SqliteConnectOptions::new()
+            .filename(&database_path)
+            .create_if_missing(true)
+            .log_statements(LevelFilter::Trace);
+
+        let connection_pool = sqlx::sqlite::SqlitePoolOptions::new()
+            .connect_with(opts)
+            .await?;
+
+        set_file_permission_owner_rw(&database_path)
+            .map_err(
+                |source| PendingCredentialRequestsStorageError::FilePermissions {
+                    path: database_path.as_ref().to_path_buf(),
+                    source,
+                },
+            )
+            .inspect_err(|err| {
+                tracing::error!("Failed to set file permissions: {err:?}");
+            })
+            .ok();
+
+        sqlx::migrate!("./migrations").run(&connection_pool).await?;
+
+        Ok(Self {
+            storage_manager: SqliteZkNymRequestsStorageManager::new(connection_pool),
+            database_path: database_path.as_ref().to_path_buf(),
+        })
+    }
+
+    pub(crate) async fn reset(&mut self) -> Result<(), PendingCredentialRequestsStorageError> {
+        // First we close the storage to ensure that all files are closed
+        self.storage_manager.close().await;
+
+        // Calling close on the storage should be enough to ensure that all files
+        // are closed but just to be sure we wait a bit
+        tokio::time::sleep(Duration::from_secs(1)).await;
+
+        // Then we remove the database file
+        std::fs::remove_file(&self.database_path)
+            .inspect_err(|err| {
+                tracing::error!("Failed to remove file: {err:?}");
+            })
+            .ok();
+
+        // Finally we recreate the storage
+        let new_storage_manager = Self::init(&self.database_path).await?;
+
+        self.storage_manager = new_storage_manager.storage_manager;
+        self.database_path = new_storage_manager.database_path.clone();
```

Comment:
why would have the path changed?

actually... would it make sense to, `on_reset`, create db with explicitly different name?

Reply:
Hmm, possibly, not sure. The whole logic for handling logout in a robust way is tricky I find. Removing the explicit database_path assignment and setting `self` instead